### ISSUE TITLE
Duplicated files

### ISSUE THREAD

**Issue by CesarHerreraG** [ubports/filemanager-app#2]
If you have a PDF file, for example in the SD, when you click on it and choose Document Viewer you'll see that it has been created a copy in /home/phablet/Documents.  If you click a SD video file and open it with Media Player you can see a copy of it in /home/phablet/Videos.

**Comment by Flohack74** [comment]
@mardy any idea why this could happen?

**Comment by sverzegnassi** [comment]
That's the standard way core apps integrate ContentHub functionalities.
The rationale is that each content type should be handle from a single - always available - place and by a single content peer. File Manager does not respect these prerequisites.

Therefore we have:
- Music type: music-app $HOME/Music
- Document type: docviewer-app $HOME/Documents
- Picture type: gallery-app $HOME/Pictures
- Video type: gallery-app $HOME/Videos

File manager really falls out of this scenario. It is not a trusty content provider, since it's a general purpose application and doesn't enforce security policies. Also, it has never been shipped in official images, as it was meant for an advanced, "hackish" usage.

Last time we core app developers have been discussing about this issue with Canonical, it has been told that File Manager should use UrlHandler for opening files from user's folders.
If a user tries to open a file that's stored in the SD card, that file should be copied into user's folders in order to make it permanently available to the system.

Long story short: this has been a design choice. :)

See for reference the ContentHub integration used in DocViewer
[DocviewerUtils](http://bazaar.launchpad.net/~ubuntu-docviewer-dev/ubuntu-docviewer-app/lo-viewer/view/head:/src/plugin/file-qml-plugin/docviewerutils.cpp)
[ContentHubProxy](http://bazaar.launchpad.net/~ubuntu-docviewer-dev/ubuntu-docviewer-app/lo-viewer/view/head:/src/app/qml/common/ContentHubProxy.qml)

**Comment by CesarHerreraG** [comment]
As File Manager does not respect the norms of ContentHub it is better delete the copy in order to be transparent to the user.

**Comment by sverzegnassi** [comment]
Document Viewer, for instance, can not access anything else except:
- $HOME/Documents
- $SDCARD/Documents

The same applies for all the other core apps that are registered as default peers in ContentHub. This has been done for security reasons: in the case of DocViewer[1] we provide a full version of LibreOffice which could contain security bugs and allow execution of non-trusted code.

The easiest way to solve this issue would be to make core apps running fully unconfined, and totally skip the ContentHub bits. This would potentially open a large crack in the whole security infrastructure.

Since the current UBports images are still based on Canonical specs, I'd rather say that File Manager shouldn't be preinstalled, since it suggests that FM is the preferable way to handle content (and it's not true, yet).

This is not an issue in File Manager itself, neither in any of those Core Apps mentioned above. A more reasonable approach would require ContentHub to inform core apps[2] about the original location of the imported file, so that e.g. DocViewer could take the proper action.

This would virtually work with any file that comes from user's XDG folder or the SD card. Access to other apps content or system files should always be confined as usual.

[1] - I've been its maintainer and main developer for years.
[2] - Core apps only, as we can't trust third party apps

**Comment by CesarHerreraG** [comment]
If the files are not deleted they could be in a subfolder as /home/phablet/Documents/COPIES
You can delete them yourself without the risk of delete other files too.

**Comment by sverzegnassi** [comment]
> If the files are not deleted they could be in a subfolder as /home/phablet/Documents/COPIES
> You can delete them yourself without the risk of delete other files too.

I prepared a patch for importing files in a subfolder more than an year ago[1], but Canonical never released such improvement[2].
Sadly, forcing an automatic removal of files in $HOME/Documents/Imported would also remove files that our users wanted to be there (e.g. by exporting a PDF with Dekko). So this really needs to be design wisely.

As a temporary workaround, though, making manual removal easier would actually work. :)

===
[1] https://code.launchpad.net/~verzegnassi-stefano/ubuntu-docviewer-app/content-hub-imported-folder/+merge/291923
[2] I've pinged them for months, I'd say it's fully their own responsability towards users who paid for their phone

**Comment by bqphone** [comment]
So @sverzegnassi are you working on solving this problem? Also if a select all option will be added, maybe cut option also. When i was moving music from phone to sd card, man that was alot of work without select all and cut option.

**Comment by sverzegnassi** [comment]
Hello @bqphone , thanks for joining the conversation!

Please open new reports for the issues you're reporting.
File system operations like 'cut' and 'select all' are not well supported by File Manager, which ultimately would need a maintainer in order to address the (several) weaknesses currently existing in its code base[1].

About your request - No, I'm not currently working on this.
The "duplicated files" issue is not so trivial as it might appear. What we're trying to fix here is not an "ordinary bug", but a design limitation.
Any attempt to solve this issue would need: 1) a valuation of the different strategies we could adopt, and to pick the one which receive the most consensus; 2) to identify the needed changes to the superstructure we have set up for preventing painful regressions to the various components of the OS.

===
[1] @Flohack74 - We might want to write down some sort of File Manager Masterplan for helping whoever could be interested in improving the app, addressing the major issues. What do you think? Could we follow up this on Telegram?

**Comment by Flohack74** [comment]
@sverzegnassi Yes ping me. Definitely this is not a "File Manager" right now, its rather a "File Viewer" only ;)

**Comment by CesarHerreraG** [comment]
The apps used to open each kind of file could be stabished in a config file that you could modify.

**Comment by nfsprodriver** [comment]
Is this issue related to the File Manager at all?

**Comment by CesarHerreraG** [comment]
I think so, but I don't be an expert. When you click on a file with filemanager it would be good the correct app oppened tht file. This don't allways happens. Thanks.

**Comment by UniversalSuperBox** [comment]
Moved to https://gitlab.com/ubports/apps/filemanager-app/-/issues/2
